samza-core/src/main/java/org/apache/samza/config/TaskConfigJava.java

```diff
@@ -37,7 +37,7 @@
 public class TaskConfigJava extends MapConfig {
   // Task Configs
   private static final String TASK_SHUTDOWN_MS = "task.shutdown.ms";
-  public static final long DEFAULT_TASK_SHUTDOWN_MS = 5000L;
+  public static final long DEFAULT_TASK_SHUTDOWN_MS = 30000L;
```

Comment:
Yes, we should make it private. Will do.
Yes, in the future we need to make sure that if the new Job model doesn't change for this specific processor, it shouldn't stop at all. And that should be majority of the case.